app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/JunkBottomSheetDialog.kt

```diff
@@ -0,0 +1,79 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.ui.main.thread.actions
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.activityViewModels
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import com.infomaniak.mail.R
+import com.infomaniak.mail.data.models.Folder.FolderRole
+import com.infomaniak.mail.data.models.message.Message
+import com.infomaniak.mail.databinding.BottomSheetJunkBinding
+import com.infomaniak.mail.ui.MainViewModel
+import com.infomaniak.mail.utils.notYetImplemented
+
+class JunkBottomSheetDialog : BottomSheetDialogFragment() {
+
+    lateinit var binding: BottomSheetJunkBinding
+    private val navigationArgs: JunkBottomSheetDialogArgs by navArgs()
+    private val mainViewModel: MainViewModel by activityViewModels()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        return BottomSheetJunkBinding.inflate(inflater, container, false).also { binding = it }.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) = with(binding) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val threadUid = navigationArgs.threadUid
+        val messageUid = navigationArgs.messageUid
+
+        if (messageUid == null) {
+            handleButtons(threadUid)
+        } else {
+            mainViewModel.getMessage(messageUid).observe(viewLifecycleOwner) { message ->
+                handleButtons(threadUid, message)
+            }
+        }
+    }
+
+    private fun handleButtons(threadUid: String, message: Message? = null) = with(binding) {
+        setSpamUi(message)
+
+        spam.setClosingOnClickListener { mainViewModel.toggleSpamOrHam(threadUid, message) }
+        phishing.setClosingOnClickListener { notYetImplemented() }
+        blockSender.setClosingOnClickListener { notYetImplemented() }
+    }
+
+    private fun setSpamUi(message: Message?) {
+        val isSpam = message?.isSpam ?: mainViewModel.isCurrentFolderRole(FolderRole.SPAM)
+        binding.spam.setText(if (isSpam) R.string.actionNonSpam else R.string.actionSpam)
+    }
+
+    private fun ActionItemView.setClosingOnClickListener(callback: (() -> Unit)) {
+        setOnClickListener {
+            callback()
+            findNavController().popBackStack()
```

Comment:
```suggestion
            findNavController().popBackStack()
            callback()
```

Calling popBackStack after the callback will close any dialog or fragment callback could navigate to. For example the future dialog for phishing confirmation

Reply:
You can also use this with setClosingOnClickListener to navigate to JunkBottomSheet without backNavigationResults